### PR TITLE
External Network: Wireguard Templates

### PR DESCRIPTION
**Pull request by cheina97**
# Description

This PR introduces the wireguard server and client templates for the new external network.

### CODE REVIEW

File: deployments/liqo/files/liqo-wireguard-gateway-client-template.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+metadata:
+  labels:
+    app: "{{ .Name }}"
```
Comment (aleoli) [inline deployments/liqo/files/liqo-wireguard-gateway-client-template.yaml:3]: we should probably define other label than "app"

File: deployments/liqo/files/liqo-wireguard-gateway-server-template.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      labels:
+        app: "{{ .Name }}"
+    spec:
+      selector:
+        app: "{{ .Name }}"
+      type: "{{ .Spec.Endpoint.ServiceType }}"
+      ports:
+      - port: "{{ .Spec.Endpoint.Port }}"
+        protocol: UDP
+        nodePort: 31820
```
Comment (aleoli) [inline deployments/liqo/files/liqo-wireguard-gateway-server-template.yaml:16]: we should not hardcode the port, but let Kubernetes take a random one

File: deployments/liqo/templates/liqo-wireguard-gateway-client-template.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+metadata:
+  name: {{ $templateConfig.name  }}
+spec:
+  objectKind:
+    apiVersion: networking.liqo.io/v1alpha1
+    kind: WgGatewayClient
+  template:
+    metadata:
+      {{- include "liqo.metadataTemplate" $templateConfig | nindent 6 }}
+    spec:
```
Comment (cheina97) [inline deployments/liqo/templates/liqo-wireguard-gateway-client-template.yaml:16]: We need it?

File: deployments/liqo/templates/liqo-wireguard-gateway-server-template.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+metadata:
+  name: {{ $templateConfig.name }}
+spec:
+  objectKind:
+    apiVersion: networking.liqo.io/v1alpha1
+    kind: WgGatewayServer
+  template: 
+    metadata:
+      {{- include "liqo.metadataTemplate" $templateConfig | nindent 6 }}
+    spec:
```
Comment (cheina97) [inline deployments/liqo/templates/liqo-wireguard-gateway-server-template.yaml:16]: We need it?

**Comment by cheina97** [comment]
It depends on https://github.com/liqotech/liqo/pull/2077

File: deployments/liqo/templates/liqo-wireguard-gateway-client-template.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                args:
+                - --name={{"{{ .Name }}"}}
+                - --namespace={{"{{ .Namespace }}"}}
+                - --remote-cluster-id={{"{{ .ClusterID }}"}}
+                - --mode=client
+                - --metrics-address=:8080
+                - --health-probe-bind-address=:8081
+                - --ping-loss-threshold=5
+                - --ping-interval=2s
+                - --ping-update-status-interval=10s
```
Comment (aleoli) [inline deployments/liqo/templates/liqo-wireguard-gateway-client-template.yaml:46]: Can helm template these values?

File: deployments/liqo/templates/liqo-wireguard-gateway-server-template.yaml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                args:
+                - --name={{"{{ .Name }}"}}
+                - --namespace={{"{{ .Namespace }}"}}
+                - --remote-cluster-id={{"{{ .ClusterID }}"}}
+                - --mode=server
+                - --metrics-address=:8080
+                - --health-probe-bind-address=:8081
+                - --ping-loss-threshold=5
+                - --ping-interval=2s
+                - --ping-update-status-interval=10s
```
Comment (aleoli) [inline deployments/liqo/templates/liqo-wireguard-gateway-server-template.yaml:57]: Here again

File: pkg/liqo-controller-manager/external-network/client-operator/client_controller.go [hunk 7]
```diff
@@ -76,8 +76,8 @@ func NewClientReconciler(cl client.Client, dynClient dynamic.Interface,
 
 // Reconcile manage GatewayClient lifecycle.
 func (r *ClientReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
-	gwClient := &networkingv1alpha1.GatewayClient{}
-	if err = r.Get(ctx, req.NamespacedName, gwClient); err != nil {
+	client := &networkingv1alpha1.GatewayClient{}
```
Comment (aleoli) [inline pkg/liqo-controller-manager/external-network/client-operator/client_controller.go:79]: Can we keep the gwClient name? Client can be misleading

File: pkg/liqo-controller-manager/external-network/client-operator/client_controller.go [hunk 8]
```diff
@@ -106,28 +106,28 @@ func (r *ClientReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 }
 
 // EnsureGatewayClient ensures the GatewayClient is correctly configured.
-func (r *ClientReconciler) EnsureGatewayClient(ctx context.Context, gwClient *networkingv1alpha1.GatewayClient) error {
-	if gwClient.Labels == nil {
-		gwClient.Labels = map[string]string{}
+func (r *ClientReconciler) EnsureGatewayClient(ctx context.Context, client *networkingv1alpha1.GatewayClient) error {
```
Comment (aleoli) [inline pkg/liqo-controller-manager/external-network/client-operator/client_controller.go:109]: Here again

File: pkg/liqo-controller-manager/external-network/server-operator/server_controller.go [hunk 9]
```diff
@@ -148,49 +148,81 @@ func (r *ServerReconciler) EnsureGatewayServer(ctx context.Context, server *netw
...
+		}
+
+		var objectTemplateMetadataAnnotations interface{}
+		if objectTemplateMetadataAnnotations, ok = objectTemplateMetadata["annotations"]; ok {
+			annotations, err := enutils.RenderTemplate(objectTemplateMetadataAnnotations, td)
+			if err != nil {
+				return fmt.Errorf("unable to render the template annotations: %w", err)
+			}
+			objChildMetadata["annotations"] = annotations
+		} */
```
Comment (aleoli) [inline pkg/liqo-controller-manager/external-network/server-operator/server_controller.go:207]: Is this commented code required?

File: pkg/liqo-controller-manager/external-network/client-operator/client_controller.go (hunk 7 above)
Comment (fra98) [reply pkg/liqo-controller-manager/external-network/client-operator/client_controller.go:79]: If we refactor to `gwClient`, I would also refactor the server_controller.go file using `gwServer` to be coherent and to ease spotting differences between the two files.

**Pull request merged**